### ISSUE TITLE
Returns zero images

### ISSUE THREAD

**Issue by bravokiloecho** [vdemedes/node-google-images#10]
I was using this package successfully earlier this year. And now I've started using it again (with the same app and with no changes to the code) but it seems to have stopped returning images.

I'm running this script:

```
googleSearch.search( 'query', function ( err, images ) {
    ...
});
```

But `images` is always returned as an empty array and `err` is returned as `false`. Is it still working for anyone else? I can't think why it would have just stopped.

I've updated the package to the latest version: `0.1.4`

**Comment by genbtc** [comment]
I Believe the API was shut down by google today 12/2 : 
{ responseData: null,
  responseDetails: 'This API is no longer available.',
  responseStatus: 403 }

Please help, i don't want to have to use the fully complicated regular API.

**Comment by vdemedes** [comment]
Oh shit. 

@genbtc is there a new API for that? Is it free?

**Comment by bravokiloecho** [comment]
Oh that's not fun.

I'd be very happy to hear of other routes...

**Comment by cpradio** [comment]
You have to use CSE, that works, but I can't find a way to make it search everything like the generic API did... it will only search the URLs I put in :(

**Comment by cpradio** [comment]
Just figured out how to get CSE to search everywhere, it is a drop down next to Sites to Search

**Comment by vdemedes** [comment]
@cpradio what's CSE?

So did anybody come up with a solution or has ideas?

**Comment by cpradio** [comment]
CSE is Custom Search API
https://developers.google.com/custom-search/

You have to signup get a code and API key, and then you can set the HUBOT_GOOGLE_CSE_KEY and HUBOT_GOOGLE_CSE_ID and it will start working again

**Comment by vdemedes** [comment]
I'm wondering if there's an API, which does not require registration and API key. Trying to find a smooth way to update everyone, without breaking their code and requiring to do additional steps.

**Comment by cpradio** [comment]
You can try to setup a generic API, but Google may shut that down. I tried to find some other services offering image search, but didn't find much

**Comment by vdemedes** [comment]
@cpradio would you be interested in working on a PR to convert this project to Custom Search API?

**Comment by cpradio** [comment]
Okay, well this is now awkward, as I could have sworn yours already did. I must have pulled mine from a different project.
https://github.com/cpradio/sp-team-jarvis/blob/master/scripts/google-images.coffee

There it is working with CSE.  Granted that is a hubot implementation, so there are slight differences. If I get a chance I may look at what it would take to get yours working.

**Comment by vdemedes** [comment]
Haha, I was wondering what HUBOT_* env variables were doing there.

Users of this project and me would really appreciate it, thanks! If you won't have time, no worries ;)

**Comment by cpradio** [comment]
Okay, so I'm not all that familiar with this project, but here are the key changes.

The following needs added about options.page = 0...

```
	options.cse_id = '' if not options.cse_id?
	options.cse_api_key = '' if not options.cse_api_key?
```

Then the request becomes

```
request "https://www.googleapis.com/customsearch/v1?q=#{ encodeURIComponent(query.replace(/\s/g, '+')) }&searchType=image&start=#{ options.page }&imgSize=#{ options.size }&cx=#{ options.cse_id }&key=#{ options.cse_api_key }", (err, res, body) ->
```

I'm not sure if the responseData will need to be altered, but that is the general gist of the changes. From there on, each user will need to pass in their cse_id and cse_api_key via the options object.

**Comment by vdemedes** [comment]
@cpradio Thank you for your help! I'll check it out soon ;)

**Comment by genbtc** [comment]
:(  i have already migrated my bot to CSE, unfortunately its only 100 queries per day for free. Really sad that your way stopped working. Took me a long time to figure CSE out, having never used google's API's before.

**Comment by dadambickford** [comment]
We used this for a slack bot to return images, mostly just commenting to keep tabs on the fix. This was a perfect solution before the change.

**Comment by vdemedes** [comment]
I understand your frustration guys, because I am frustrated as well. I did some research and there's no easy way to continue using google images API. Other APIs are hard to get started with.

The only viable alternative is Flickr API. I tried their search API method and it seems to work good. I recommend using [flickrapi](https://www.npmjs.com/package/flickrapi) module from now on. I haven't used it myself, but it looks good.

Thank you for using `google-images` module, it was a long time, now it's time to say goodbye :cry:
